File: front-end/Nova/nova-core/HelloWorldPlugin.ts
import NovaPlugin from './abstractPlugin';

export default class HelloWorldPlugin extends NovaPlugin {

    getKeywords(): string[] {
        return ['Hello', 'Hi', 'Howdy'];
    }

    execute(command: string): string {
        return 'Hello! My name is Nova.';
    }

    executeSecondaryCommnand(command: string): string {
        return 'To teach me more fun things to do, go to the plugin store.';
    }

}

File: nova-core/plugins/hello_world_plugin.py
from core.abstract_plugin import NovaPlugin

class HelloWorldPlugin(NovaPlugin):

    def get_keywords(self) -> list:
        return ['hello', 'hi', 'hey']

    def execute(self, command: str) -> str:
        if ('hello' in command, 'hi' in command or 'hey' in command):
            return 'Hello! My name is Nova. How can I help you?'
        
        return None

    def help_command(self, command: str) -> str:
        return 'Helping the world by helping you.'